pkg/monitors/nagios/nagios.go

```diff
@@ -0,0 +1,253 @@
+package nagios
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/go-test/deep"
+	"github.com/kballard/go-shellquote"
+	"github.com/patrickmn/go-cache"
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/golib/v3/event"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `singleInstance:"false" acceptsEndpoints:"false"`
+	// The command to exec with any arguments like:
+	// `"LC_ALL=\"en_US.utf8\" /usr/lib/nagios/plugins/check_ntp_time -H pool.ntp.typhon.net -w 0.5 -c 1"`
+	Command string `yaml:"command" validate:"required"`
+	// Corresponds to the nagios `service` column and allows to aggregate all
+	// instances of the same service (when calling the same check script with
+	// different arguments)
+	Service string `yaml:"service" validate:"required"`
+	// The max execution time allowed in seconds be fore to send SIGTERM. In
+	// any case, a SIGKILL will be send at `intervalSeconds` value.
+	Timeout int `yaml:"timeout" default:"9"`
+}
+
+// Monitor that collect metrics
+type Monitor struct {
+	Output types.FilteringOutput
+	cancel func()
+	logger logrus.FieldLogger
+}
+
+const (
+	unknown          = 3
+	cacheKey         = "lastRun"
+	propertiesLength = 256
+)
+
+// Configure and kick off internal metric collection
+func (m *Monitor) Configure(conf *Config) error {
+	m.logger = logrus.WithFields(logrus.Fields{"monitorType": monitorType})
+	// Define global dimensions used for both datapoint and event
+	dimensions := map[string]string{
+		"plugin":  "nagios",
+		"command": conf.Command,
+		"service": conf.Service,
+	}
+	// Enforce interval greater than command timeout
+	if conf.IntervalSeconds < conf.Timeout {
+		return fmt.Errorf("configured timeout must be lower than intervalSeconds")
+	}
+	// Init cache used to avoid sending duplicate event for each interval
+	c := cache.New(1*time.Minute, 1*time.Hour)
+
+	// Start the metric gathering process here
+	var ctx context.Context
+	ctx, m.cancel = context.WithCancel(context.Background())
+	utils.RunOnInterval(ctx, func() {
+		// Run command
+		stdout, stderr, err := runCommand(conf.Command, conf.Timeout, conf.IntervalSeconds)
+		state, err := getExitCode(err, stdout)
+
+		// Send datapoint with only state
+		m.Output.SendDatapoints([]*datapoint.Datapoint{
+			datapoint.New(
+				nagiosState,
+				dimensions,
+				datapoint.NewIntValue(int64(state)),
+				datapoint.Gauge,
+				time.Time{}),
+		}...)
+
+		properties, diffProperties := makeProperties(state, err, stdout, stderr)
+		m.logger.Warn(properties)
```

Comment:
Remove these or make Debug levels.

Reply:
sorry it was a debug test I forgot in the last commit ..